src/openforms/appointments/renderer.py

```diff
@@ -0,0 +1,56 @@
+from dataclasses import dataclass
+from typing import Any, Iterator
+
+from openforms.formio.rendering.nodes import ComponentNodeBase
+from openforms.formio.service import FormioData, format_value
+from openforms.submissions.rendering import Renderer
+
+from .service import get_appointment
+
+
+@dataclass
+class ContactDetailNode(ComponentNodeBase):
+    """
+    Rendering of a contact detail formio component.
+
+    Ideally we'd be able to share more logic with the
+    :class:`openforms.formio.rendering.nodes.ComponentNode`, but that is too tighly
+    coupled with ``FormStep`` and ``SubmissionStep`` Django models at the moment.
+
+    Therefore, this implementation is extremely minimal.
+    """
+
+    data: FormioData
+
+    @property
+    def value(self) -> Any:
+        assert "key" in self.component
+        return self.data[self.component["key"]]
```

Comment:
yeah mixed feelings - it's a type guard because the keys in the `Component` are all marked as optional until we can upgrade to Python 3.11. Components without key will blow up anyway.